app/main.py

```diff
@@ -1,2 +1,76 @@
+from typing import Any, Hashable
+
+
 class Dictionary:
-    pass
+
+    def __init__(self) -> None:
+        self._size = 8
+        self._table = [None] * self._size
+        self._counter = 0
+
+    class Node:
+        def __init__(self, key: Hashable, value: Any) -> None:
+            self.key = key
+            self.value = value
+            self._hash = hash(key)
+
+        def __str__(self) -> str:
+            key_string = \
+                f"'{self.key}'" if isinstance(self.key, str) \
+                else self.key
+            value_string = \
+                f"'{self.value}'" if isinstance(self.value, str) \
+                else self.value
+            return f"{key_string}: {value_string}"
+
+    def _resize(self) -> None:
+        old_table = self._table
+        self._size *= 2
+        self._table = [None] * self._size
+        self._counter = 0
+
+        for node in old_table:
+            if node:
+                index = self._hash_to_index(node.key)
```

Comment:
The `_resize` method should rehash the nodes using the new table size. Currently, it uses the old hash values, which can lead to incorrect behavior. You should call `self._hash_to_index(node.key)` after updating the `_size` to ensure the correct index is calculated for the resized table.